pkg/controller/installation/core_controller.go

```diff
@@ -748,6 +774,31 @@ func (r *ReconcileInstallation) Reconcile(ctx context.Context, request reconcile
 		return reconcile.Result{}, err
 	}
 
+	// See the section 'Node and Installation finalizer' at the top of this file for details.
+	if terminating {
+		// Keep the finalizer on the Installation until the ClusterRoleBinding for calico-node
+		// (and ClusterRole and ServiceAccount) is removed.
+		crb := rbacv1.ClusterRoleBinding{}
+		crbKey := types.NamespacedName{Name: "calico-node"}
+		err := r.client.Get(ctx, crbKey, &crb)
+		if err != nil && !apierrors.IsNotFound(err) {
+			r.SetDegraded("Unable to get ClusterRoleBinding", err, reqLogger)
+			return reconcile.Result{}, err
+		}
+		found := false
+		for _, x := range crb.Finalizers {
```

Comment:
Thinking on this, it seems pretty unnecessary to check for the finalizer. If the finalizer is remove then the resource will be deleted.